core-bundle/src/EventListener/SwitchUserListener.php

```diff
@@ -59,9 +60,25 @@ public function __invoke(SwitchUserEvent $event): void
             $targetUser = $targetUser->getUsername();
         }
 
-        $this->logger->info(
-            sprintf('User "%s" has switched to user "%s"', $sourceUser, $targetUser),
-            ['contao' => new ContaoContext(__METHOD__, ContaoContext::ACCESS, $sourceUser)]
-        );
+        $originalUser = null;
+
+        if ($token instanceof SwitchUserToken) {
+            $originalUser = $token->getOriginalToken()->getUser()->getUsername();
+        }
+
+        if ($originalUser === $targetUser) {
+            $this->logger->info(
+                sprintf('User "%s" has quit the impersonation of user "%s"', $originalUser, $sourceUser),
+                ['contao' => new ContaoContext(__METHOD__, ContaoContext::ACCESS, $originalUser)]
+            );
+        } else {
+            if (!empty($originalUser)) {
```

Comment:
Is there a special reason to use `!empty` instead of checking for `null !== $originalUser`?

Reply:
No, I think both is good. The `empty` is more or less my general choice of sting comparing, because it includes `null` and empty sting.

Reply:
👍